WalletWasabi/WabiSabi/Client/Banning/CoinPrison.cs

```diff
@@ -0,0 +1,116 @@
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.WabiSabi.Client.Banning;
+
+public class CoinPrison
+{
+	public CoinPrison(string filePath)
+	{
+		FilePath = filePath;
+	}
+
+	public List<PrisonedCoinRecord> BannedCoins { get; set; } = new();
+	public string FilePath { get; set; }
+	private object Lock { get; set; } = new();
+
+	public bool TryGetOrRemoveBannedCoin(SmartCoin coin, DateTimeOffset banDeadlineTime, [NotNullWhen(true)] out DateTimeOffset? bannedUntil)
```

Comment:
> Yes, good argument. Lucas suggested that it could get the end of input registration so we know if the coin can be registered as an input for the next coinjoin, but I don't see where to get that timestamp from.

It could get the end of input registration, but it doesn't. Neither in this implementation nor in any other.

I don't understand why we are preparing this function for a feature, which is not implemented (may never be).

In order to do that, we need to do the CoinPrison checking after we found the `Round`.
This can be done either in `CjClient` or in `CoinJoinCoinSelector`. But IMHO the drawbacks and complexity outweight the benefits. 

My two sats.

This is not a blocker, but it raises a question in my mind every time I see it.

Reply:
I understand your point and I can stand behind it. I kept it like this because this was the suggestion before to go with, but I'm still open for dropping the hardcoded parameter call, and just use it inside.